### PR TITLE
use new build-and-push action

### PR DESCRIPTION
**Pull request by zmiklank**
Using the new build-and-push-action [1].

[1] https://github.com/sclorg/build-and-push-action

### CODE REVIEW

File: .github/workflows/build-and-push.yml [hunk 1]
```diff
@@ -1,99 +1,42 @@
...
-      - name: Print image url
-        if: ${{ matrix.registry_namespace == 'centos7' }}
-        run: echo "Image pushed to ${{ steps.push-to-quay-centos7.outputs.registry-paths }}"
-
-      - name: Print image url
-        if: ${{ matrix.registry_namespace != 'centos7' }}
-        run: echo "Image pushed to ${{ steps.push-to-quay-sclorg.outputs.registry-paths }}"
+          registry: "quay.io"
+          registry_namespace: ${{ matrix.registry_namespace }}
+          registry_username: ${{ secrets[matrix.quayio_username] }}
```
Comment (phracek) [inline .github/workflows/build-and-push.yml:38]: I would prefer to not add secrets into the matrix but split it into two actions.
One for centos7 and one for c9s based on the matrix.tag.

But this is an idea. WDYT? Does it make sense to you?

File: .github/workflows/build-and-push.yml [hunk 2]
```diff
@@ -1,99 +1,42 @@
...
+          registry_token: ${{ secrets[matrix.quayio_token] }}
```
Comment (phracek) [inline .github/workflows/build-and-push.yml:39]: I thinks the format should be:
${{ secrets.[matrix.quayio_token] }} right?
I guess, '.' is missing.

**Review by phracek — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-push.yml (hunk 1 above)
Comment (phracek) [reply .github/workflows/build-and-push.yml:38]: But this is not needed. Based on the later comment. Sorry.

File: .github/workflows/build-and-push.yml (hunk 2 above)
Comment (zmiklank) [reply .github/workflows/build-and-push.yml:39]: I was inspired by [this article](https://sbulav.github.io/terraform/github-actions-matrix-secrets/).
The dot is not used there.

Comment (phracek) [reply .github/workflows/build-and-push.yml:39]: I see. ok.

**Pull request merged**